builds/fabric8-che/plugins/keycloak-plugin-server/src/main/java/com/redhat/che/keycloak/server/KeycloakAuthenticationFilter.java

```diff
@@ -15,14 +15,22 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             throws IOException, ServletException {
         HttpServletRequest request = (HttpServletRequest) req;
         String auth = request.getHeader("Authorization");
-        if(auth == null ){
+        String uri = request.getRequestURI();
+
+        if (uri.endsWith("/api/system/state")) {
```

Comment:
+1 to have two separate methods: `isWebSocketRequest()` and `isNonSecuredEndpoint()`